graylog2-server/src/main/java/org/graylog2/configuration/ExposedConfiguration.java

```diff
@@ -0,0 +1,117 @@
+package org.graylog2.configuration;
+
+import com.google.common.collect.ImmutableList;
+import org.graylog2.Configuration;
+import org.graylog2.rest.models.system.configuration.ConfigurationVariable;
+
+/**
+ * List of configuration values that are safe to return, i.e. do not include any sensitive
+ * information. Building a list manually because we need to guarantee never to return any
+ * sensitive variables like passwords etc. - See this as a whitelist approach.
+ */
+public class ExposedConfiguration {
+
+    private final int inputBufferProcessors;
+    private final int processBufferProcessors;
+    private final int outputBufferProcessors;
+
+    private final String processorWaitStrategy;
+    private final String inputBufferWaitStrategy;
+    private final int inputBufferRingSize;
+    private final int ringSize;
+    private final String pluginDir;
+    private final String nodeIdFile;
+
+    private final boolean allowHighlighting;
+    private final boolean allowLeadingWildcardSearches;
+
+    private final String rotationStrategy;
+    private final String retentionStrategy;
+
+    private final int maxDocsPerIndex;
+    private final long maxSizePerIndex;
+    private final String maxTimePerIndex;
+    private final int maxNumberOfIndices;
+    private final int shards;
+    private final int replicas;
+
+    private final long streamProcessingTimeout;
+    private final int streamProcessingMaxFaults;
+    private final long outputModuleTimeout;
+    private final int staleMasterTimeout;
+
+    private final boolean disableIndexOptimization;
+    private final int indexOptimizationMaxSegments;
+
+    private final String gcWarningThreshold;
+
+    public ExposedConfiguration(Configuration configuration, ElasticsearchConfiguration esConfiguration) {
```

Comment:
`org.graylog2.rest` cannot access the classes in `org.graylog2.configuration` so the `ExposedConfiguration` class is required to map configuration values to a `List`. We'd have to do this in the JAX-RS logic and that smells.

Given how close the code freeze is and how much this small change is improving our support process I'd ask you to merge it the way it is if you don't have strong opinions against it except a preference for not using this class and moving the logic.

You can of course also perform the change yourself and then merge if you see an easier way that makes sense.

Thank you! :)